package Testbed.proto;

message TestGameComponent1 {
	optional string entryName = 1;
	optional uint64 entryOffset = 2;
	optional uint64 entrySize = 3;
	optional string entryType = 4;
}

message TestGameComponent2 {
	repeated string			packageDependencies = 1;
	optional uint64			highscore = 2;
}

message Highscore {
	optional string name = 1;
	optional uint64 highscore = 2;
}

message GameHighscores {
	repeated Highscore highscores = 1;
	optional TestGameComponent2 extra1 = 2;
	optional int32 extra2 = 3;
}